src/test/kotlin/io/github/gradlenexus/publishplugin/TaskOrchestrationTest.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import org.assertj.core.api.Assertions.assertThat
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.testfixtures.ProjectBuilder
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.nio.file.Path
+
+class TaskOrchestrationTest {
+
+    @TempDir
+    lateinit var projectDir: Path
+
+    private lateinit var project: Project
+
+    @BeforeEach
+    internal fun setUp() {
+        project = ProjectBuilder.builder().withProjectDir(projectDir.toFile()).build()
+    }
+
+    @Test
+    internal fun `close task should run after init`() {
+        initSingleProjectWithDefaultConfiguration()
+        assertGivenTaskMustRunAfterAnother("closeSonatypeStagingRepository", "initializeSonatypeStagingRepository")
+    }
+
+    @Test
+    @Disabled("Broken - must run only after publishMavenJavaPublicationToSonatypeRepository not publishToSonatypeRepository")
+    internal fun `close task should run after related publish`() {
+        initSingleProjectWithDefaultConfiguration()
+        assertGivenTaskMustRunAfterAnother("closeSonatypeStagingRepository", "publishToSonatypeRepository")
```

Comment:
publishToNexusTask (and close/release tasks) depends/mustRunAfter publishMavenJavaPublicationToSonatypeRepository. It would be probably better to use publishToSonatypeRepository to also cover PDF documentation update, CDC stubs, etc. @marcphilipp How would be tweak it to cover that?

Reply:
I'm afraid I don't understand your question. Could you please elaborate a bit more?

Reply:
Sure. How the easiest would be to change the plugin implementation to make `closeSonatypeStagingRepository` depends on `publishToSonatypeRepository` instead of all the related publications separately (e.g. `publishMavenJavaPublicationToSonatypeRepository`, `publishPdfDocumentationPublicationToSonatypeRepository`)?

It's not a big deal as a publication for PDF documents should be also covered by:
```
        val publishTasks = project.tasks
                .withType<PublishToMavenRepository>()
                .matching { it.repository == nexusRepository }
```

Reply:
Can we just add `closeTask.mustRunAfter(publishToNexusTask)` and the same for `releaseTask`?

Reply:
How would be best to get a reference/provider to the umbrella publish task in the production plugin code?